iree/compiler/Dialect/Flow/Transforms/DeduplicateExecutables.cpp

```diff
@@ -0,0 +1,208 @@
+
+  void runOnOperation() override {
+    auto moduleOp = getOperation();
+    auto executableOps = llvm::to_vector<8>(moduleOp.getOps<ExecutableOp>());
+    auto builder = OpBuilder::atBlockBegin(moduleOp.getBody());
+
+    SmallVector<ExecutableOp, 3> duplicateExecutableOps;
+    DenseMap<Attribute, Attribute> entryPointRefReplacements;
+
+    // For each executable, find the first executable which it is equivalent to.
+    for (int i = executableOps.size() - 1; i >= 0; --i) {
+      auto executableOp = executableOps[i];
+
+      for (int j = 0; j < i; ++j) {
+        auto comparisonExecutableOp = executableOps[j];
+
+        if (!areExecutablesEquivalent(executableOp, comparisonExecutableOp)) {
+          continue;
+        }
+
+        // Found an equivalent executable! Record it and move on to the next.
+        duplicateExecutableOps.push_back(executableOp);
+
+        // Record entry point reference replacements.
+        auto dispatchEntryOps = llvm::to_vector<1>(
+            executableOp.getBlock().getOps<DispatchEntryOp>());
+        auto comparisonDispatchEntryOps = llvm::to_vector<1>(
+            comparisonExecutableOp.getBlock().getOps<DispatchEntryOp>());
+        assert(dispatchEntryOps.size() == comparisonDispatchEntryOps.size());
+        for (int k = 0; k < dispatchEntryOps.size(); ++k) {
+          auto oldSymbolRefAttr = builder.getSymbolRefAttr(
+              executableOp.getName(),
+              {builder.getSymbolRefAttr(dispatchEntryOps[k].sym_name())});
+          auto newSymbolRefAttr = builder.getSymbolRefAttr(
+              comparisonExecutableOp.getName(),
+              {builder.getSymbolRefAttr(
+                  comparisonDispatchEntryOps[k].sym_name())});
+          entryPointRefReplacements[oldSymbolRefAttr] = newSymbolRefAttr;
+        }
+
+        break;
+      }
+    }
+
+    totalExecutables = executableOps.size();
+    executablesDeduplicated = duplicateExecutableOps.size();
+    remainingExecutables = totalExecutables - executablesDeduplicated;
+
+    replaceEntryPointUses(moduleOp, entryPointRefReplacements);
+    for (auto executableOp : duplicateExecutableOps) {
+      executableOp.erase();
+    }
+
+    // TODO(scotttodd): rewrite executable indices, filling in gaps?
+  }
+
+ private:
+  Statistic totalExecutables{
+      this, "total executable(s)",
+      "Number of flow.executable ops before deduplication"};
+  Statistic executablesDeduplicated{
+      this, "duplicate executable(s)",
+      "Number of flow.executable ops removed as duplicates"};
+  Statistic remainingExecutables{
+      this, "unique executable(s)",
+      "Number of flow.executable ops remaining after deduplication"};
+};
+
+std::unique_ptr<OperationPass<ModuleOp>> createDeduplicateExecutablesPass() {
+  return std::make_unique<DeduplicateExecutablesPass>();
+}
+
+static PassRegistration<DeduplicateExecutablesPass> pass(
+    "iree-flow-dedupliclate-executables",
```

Comment:
an extra letter in there :P

Reply:
oops, thanks for catching! spell checker failed me here :(